source/common/grpc/async_client_impl.cc

```diff
@@ -103,7 +103,7 @@ void AsyncStreamImpl::onHeaders(Http::HeaderMapPtr&& headers, bool end_stream) {
     // https://github.com/grpc/grpc/blob/master/doc/http-grpc-status-mapping.md requires that
     // grpc-status be used if available.
     if (end_stream && grpc_status) {
-      onTrailers(std::move(headers));
+      onTrailers(std::move(headers)); // NOLINT(bugprone-use-after-move)
```

Comment:
Hmm, this seems like an actual bug if we move in the call above? @htuch?

Reply:
No, the move above only happens when end_stream==false, and here is end_stream==true.

Reply:
OK, makes sense. Add a comment? I like this check!